fineract-core/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java

```diff
@@ -437,7 +437,7 @@ public static SavingsAccountTransactionData withHoldTax(final SavingsAccountData
                 savingsAccountTransactionType.getValue().longValue(), savingsAccountTransactionType.getCode(),
                 savingsAccountTransactionType.getValue().toString());
         SavingsAccountTransactionData accountTransaction = new SavingsAccountTransactionData(amount.getAmount(), date,
-                savingsAccount.getId(), transactionType, isReversed, null, isManualTransaction, lienTransaction);
+                savingsAccount.getId(), transactionType, isReversed, null, isManualTransaction, lienTransaction, date);
```

Comment:
This looks wrong to me. We should not set the submitted on date by the transaction date.

Reply:
Same interest posting